Unable to `make install-all-tanzu-cli-plugins` off main.

## Bug Report

Off `main`, i'm unable to run `make install-all-tanzu-cli-plugins`. It appears to be failing when doing the `plugin install all *` command. This could be related to our new version of `tanzu-framework`?

```
✔  successfully built local repository
make[1]: Leaving directory '/tmp/tce-release/tanzu-framework'
+ popd
/tmp/tce-release
TCE_RELEASE_DIR=/tmp/tce-release \
FRAMEWORK_BUILD_VERSION=$(cat "./hack/FRAMEWORK_BUILD_VERSION") ENVS="linux-amd64 windows-amd64 darwin-amd64 darwin-arm64" hack/builder/install-tanzu.sh
+ [[ -z /tmp/tce-release ]]
+ ROOT_REPO_DIR=/tmp/tce-release
+ '[' '!' -d /tmp/tce-release ']'
+ '[' '!' -d /tmp/tce-release/tanzu-framework ']'
+ cd /tmp/tce-release/tanzu-framework
+++ git rev-parse --short HEAD
++ git describe --match=fd96beb --always
+ BUILD_SHA=fd96beb
+ sed -i.bak -e 's/ --dirty//g' ./Makefile
+ rm ./Makefile.bak
+ sed -i.bak -e 's|--artifacts artifacts/${OS}/${ARCH}/cli|--artifacts artifacts|g' ./Makefile
+ rm ./Makefile.bak
+ sed -i.bak -e 's|--artifacts artifacts-admin/${GOHOSTOS}/${GOHOSTARCH}/cli|--artifacts artifacts-admin|g' ./Makefile
+ rm ./Makefile.bak
+ sed -i.bak -e 's|--local $(ARTIFACTS_DIR)/$(GOHOSTOS)/$(GOHOSTARCH)/cli|--local $(ARTIFACTS_DIR)|g' ./Makefile
+ rm ./Makefile.bak
+ sed -i.bak -e 's|--local $(ARTIFACTS_DIR)-admin/$(GOHOSTOS)/$(GOHOSTARCH)/cli|--local $(ARTIFACTS_DIR)-admin|g' ./Makefile
+ rm ./Makefile.bak
+ sed -i.bak -e 's/$(shell git describe --tags --abbrev=0 2>$(NUL))/v0.10.0/g' ./Makefile
+ rm ./Makefile.bak
++ go env GOHOSTOS
+ GOHOSTOS=linux
+ [[ linux-amd64 windows-amd64 darwin-amd64 darwin-arm64 == *\l\i\n\u\x* ]]
+ BUILD_SHA=fd96beb
+ BUILD_VERSION=v0.10.0
+ TANZI_CLI_NO_INIT=true
+ TANZU_CORE_BUCKET=tce-tanzu-cli-framework
+ make 'ENVS=linux-amd64 windows-amd64 darwin-amd64 darwin-arm64' install-cli-plugins install-cli
make[1]: Entering directory '/tmp/tce-release/tanzu-framework'
which: no controller-gen in (/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl::/home/josh/s:/home/josh/.local/bin/statusbar:/usr/lib/jvm/default/bin::/home/josh/s:/home/josh/.local/bin/statusbar)
TANZU_CLI_NO_INIT=true go run -ldflags "-s -w -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.Date=$(date -u +"%Y-%m-%d")' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.SHA=fd96beb' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.Version=v0.10.0' -X 'main.BuildEdition=tkg' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.IsOfficialBuild=""' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/config.CoreBucketName=tce-tanzu-cli-framework'" ./cmd/cli/tanzu/main.go config set unstable-versions "experimental"
TANZU_CLI_NO_INIT=true go run -ldflags "-s -w -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.Date=$(date -u +"%Y-%m-%d")' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.SHA=fd96beb' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.Version=v0.10.0' -X 'main.BuildEdition=tkg' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.IsOfficialBuild=""' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/config.CoreBucketName=tce-tanzu-cli-framework'" ./cmd/cli/tanzu/main.go \
                plugin install all --local ./artifacts
Error: could not find manifest.yaml file: open artifacts/manifest.yaml: no such file or directory

✖  could not find manifest.yaml file: open artifacts/manifest.yaml: no such file or directory
exit status 1
make[1]: *** [Makefile:298: install-cli-plugins] Error 1
make[1]: Leaving directory '/tmp/tce-release/tanzu-framework'
make: *** [Makefile:276: install-cli] Error 2
```

## Expected Behavior

Builds to complete

## Steps to Reproduce the Bug

1. Clean your filesystem
2. Clone the repo
3. Run `make install-all-tanzu-cli-plugins`

## Environment Details

* Build version (`tanzu version`):  none
* Operating System (client): `Linux tugboat 5.15.13-arch1-1 #1 SMP PREEMPT Wed, 05 Jan 2022 16:20:59 +0000 x86_64 GNU/Linux`

--- a/Makefile
+++ b/Makefile
@@ -113,6 +113,10 @@ OCI_REGISTRY := projects.registry.vmware.com/tce
 .PHONY: lint mdlint shellcheck check yamllint misspell actionlint urllint imagelint
 check: ensure-deps lint mdlint shellcheck yamllint misspell actionlint urllint imagelint
 
+.PHONY: check-deps-minimum-build
+check-deps-minimum-build:
+	hack/ensure-deps/check-zip.sh
+
 .PHONY: ensure-deps
 ensure-deps:
 	hack/ensure-deps/ensure-dependencies.sh
@@ -185,20 +189,20 @@ $(TOOLING_BINARIES):
 ##### Tooling Binaries
 
 ##### BUILD TARGETS #####
-build-tce-cli-plugins: version clean-plugin build-cli-plugins ## builds the CLI plugins that live in the TCE repo into the artifacts directory
+build-tce-cli-plugins: version clean-plugin check-deps-minimum-build build-cli-plugins ## builds the CLI plugins that live in the TCE repo into the artifacts directory
 	@printf "\n[COMPLETE] built TCE-specific plugins at $(ARTIFACTS_DIR)\n"
 	@printf "To install these plugins, run \`make install-tce-cli-plugins\`\n"
 
-install-tce-cli-plugins: version clean-plugin build-cli-plugins framework-set-unstable-versions install-plugins ## builds and installs CLI plugins found in artifacts directory @printf "\n[COMPLETE] built and installed TCE-specific plugins at $${XDG_DATA_HOME}/tanzu-cli/. "
+install-tce-cli-plugins: version clean-plugin check-deps-minimum-build build-cli-plugins framework-set-unstable-versions install-plugins ## builds and installs CLI plugins found in artifacts directory @printf "\n[COMPLETE] built and installed TCE-specific plugins at $${XDG_DATA_HOME}/tanzu-cli/. "
 	@printf "These plugins will be automatically detected by your tanzu CLI.\n"	
 
-build-all-tanzu-cli-plugins: version clean build-cli build-cli-plugins ## builds the Tanzu CLI and all CLI plugins that are used in TCE
+build-all-tanzu-cli-plugins: version clean check-deps-minimum-build build-cli build-cli-plugins ## builds the Tanzu CLI and all CLI plugins that are used in TCE
 	@printf "\n[COMPLETE] built plugins at $(ARTIFACTS_DIR)\n"
 	@printf "These plugins will be automatically detected by tanzu CLI.\n"
 	@printf "\n[COMPLETE] built tanzu CLI at $(TCE_RELEASE_DIR). "
 	@printf "Move this binary to a location in your path!\n"
 
-install-all-tanzu-cli-plugins: version clean build-cli install-cli build-cli-plugins install-plugins ## installs the Tanzu CLI and all CLI plugins that are used in TCE
+install-all-tanzu-cli-plugins: version clean check-deps-minimum-build build-cli install-cli build-cli-plugins install-plugins ## installs the Tanzu CLI and all CLI plugins that are used in TCE
 	@printf "\n[COMPLETE] built and installed TCE-specific plugins at $${XDG_DATA_HOME}/tanzu-cli/.\n"
 	@printf "These plugins will be automatically detected by your tanzu CLI.\n"
 	@printf "\n[COMPLETE] built and installed tanzu CLI at $(TANZU_CLI_INSTALL_PATH). "

--- a/hack/builder/build-tanzu.sh
+++ b/hack/builder/build-tanzu.sh
@@ -23,7 +23,9 @@ fi
 ROOT_REPO_DIR="${TCE_RELEASE_DIR}"
 rm -fr "${ROOT_REPO_DIR}"
 mkdir -p "${ROOT_REPO_DIR}"
-cd "${ROOT_REPO_DIR}" || exit 1
+
+# recreate the TF repo
+pushd "${ROOT_REPO_DIR}" || exit 1
 
 if [[ -z "${TCE_BUILD_VERSION}" ]]; then
     echo "TCE_BUILD_VERSION is not set"
@@ -37,6 +39,10 @@ if [[ -n "${TANZU_FRAMEWORK_REPO_HASH}" ]]; then
 fi
 git clone --depth 1 --branch "${TANZU_FRAMEWORK_REPO_BRANCH}" "${TANZU_FRAMEWORK_REPO}" "tanzu-framework"
 set -x
+
+popd || exit 1
+
+# now build TF
 pushd "${ROOT_REPO_DIR}/tanzu-framework" || exit 1
 git reset --hard
 if [[ -n "${TANZU_FRAMEWORK_REPO_HASH}" ]]; then

--- a/hack/builder/install-tanzu.sh
+++ b/hack/builder/install-tanzu.sh
@@ -20,11 +20,12 @@ if [ ! -d "${ROOT_REPO_DIR}" ] || [ ! -d "${ROOT_REPO_DIR}/tanzu-framework" ]; t
     echo "Please run \`make build-cli\` first to clone the repository"
 fi
 
-cd "${ROOT_REPO_DIR}/tanzu-framework" || exit 1
+pushd "${ROOT_REPO_DIR}/tanzu-framework" || exit 1
 
 BUILD_SHA="$(git describe --match="$(git rev-parse --short HEAD)" --always)"
 sed -i.bak -e "s/ --dirty//g" ./Makefile && rm ./Makefile.bak
 sed -i.bak -e "s|--artifacts artifacts/\${OS}/\${ARCH}/cli|--artifacts artifacts|g" ./Makefile && rm ./Makefile.bak
+sed -i.bak -e "s|--artifacts artifacts-admin/\${OS}/\${ARCH}/cli|--artifacts artifacts-admin|g" ./Makefile && rm ./Makefile.bak
 sed -i.bak -e "s|--artifacts artifacts-admin/\${GOHOSTOS}/\${GOHOSTARCH}/cli|--artifacts artifacts-admin|g" ./Makefile && rm ./Makefile.bak
 sed -i.bak -e "s|--local \$(ARTIFACTS_DIR)/\$(GOHOSTOS)/\$(GOHOSTARCH)/cli|--local \$(ARTIFACTS_DIR)|g" ./Makefile && rm ./Makefile.bak
 sed -i.bak -e "s|--local \$(ARTIFACTS_DIR)-admin/\$(GOHOSTOS)/\$(GOHOSTARCH)/cli|--local \$(ARTIFACTS_DIR)-admin|g" ./Makefile && rm ./Makefile.bak
@@ -34,5 +35,7 @@ sed -i.bak -e "s/\$(shell git describe --tags --abbrev=0 2>\$(NUL))/${FRAMEWORK_
  #The tanzu-framework `build-install-cli-all` target always uses the current host OS, and if that's not being built it will fail.
 GOHOSTOS=$(go env GOHOSTOS)
 if [[ "$ENVS" == *"${GOHOSTOS}"* ]]; then
-    BUILD_SHA=${BUILD_SHA} BUILD_VERSION=${FRAMEWORK_BUILD_VERSION} TANZI_CLI_NO_INIT=true TANZU_CORE_BUCKET="tce-tanzu-cli-framework" make ENVS="${ENVS}" install-cli-plugins install-cli
+    BUILD_SHA=${BUILD_SHA} BUILD_VERSION=${FRAMEWORK_BUILD_VERSION} TANZI_CLI_NO_INIT=true TANZU_CORE_BUCKET="tce-tanzu-cli-framework" make ENVS="${ENVS}" install-cli install-cli-plugins
 fi
+
+popd || exit 1

--- a/hack/ensure-deps/check-curl.sh
+++ b/hack/ensure-deps/check-curl.sh
@@ -1,0 +1,22 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script can be used to check your local development environment
+# for necessary dependencies used in TCE
+
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+TCE_CI_BUILD="${TCE_CI_BUILD:-""}"
+
+if [[ "${TCE_CI_BUILD}" != "true" ]]; then
+
+if [[ -z "$(command -v curl)" ]]; then
+  echo "**** Please install curl before proceeding *****"
+  exit 1
+fi
+
+fi

--- a/hack/ensure-deps/check-zip.sh
+++ b/hack/ensure-deps/check-zip.sh
@@ -1,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script can be used to check your local development environment
+# for necessary dependencies used in TCE
+
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+TCE_CI_BUILD="${TCE_CI_BUILD:-""}"
+
+if [[ "${TCE_CI_BUILD}" != "true" ]]; then
+
+if [[ -z "$(command -v zip)" ]]; then
+  echo "**** Please install zip before proceeding *****"
+  exit 1
+fi
+if [[ -z "$(command -v unzip)" ]]; then
+    echo "**** Please install unzip before proceeding *****"
+    exit 1
+fi
+
+fi

--- a/hack/ensure-deps/ensure-curl.sh
+++ b/hack/ensure-deps/ensure-curl.sh
@@ -1,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script can be used to check your local development environment
+# for necessary dependencies used in TCE
+
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+TCE_CI_BUILD="${TCE_CI_BUILD:-""}"
+
+SUDO_CMD="sudo"
+if [[ "${TCE_CI_BUILD}" == "true" ]]; then
+  SUDO_CMD=""
+fi
+
+if [[ -z "$(command -v curl)" ]]; then
+  if [[ -f "/etc/redhat-release" ]]; then
+    ${SUDO_CMD} yum -y install curl
+  elif [[ "$(grep Ubuntu /etc/os-release)" != "" ]]; then
+    ${SUDO_CMD} apt-get -y install curl
+  else
+    echo "**** Please install curl before proceeding *****"
+    exit 1
+  fi
+fi

--- a/hack/ensure-deps/ensure-dependencies.sh
+++ b/hack/ensure-deps/ensure-dependencies.sh
@@ -37,11 +37,6 @@ if [[ -z "$(command -v docker)" ]]; then
     echo "Missing docker"
     ((i=i+1))
 fi
-
-if [[ -z "$(command -v curl)" ]]; then
-    echo "Missing curl"
-    ((i=i+1))
-fi
 # these are must have dependencies to just get going
 
 if [[ $i -gt 0 ]]; then
@@ -50,6 +45,8 @@ if [[ $i -gt 0 ]]; then
     exit 1
 fi
 
+"${TCE_REPO_PATH}/hack/ensure-deps/ensure-curl.sh"
+"${TCE_REPO_PATH}/hack/ensure-deps/ensure-zip.sh"
 "${TCE_REPO_PATH}/hack/ensure-deps/ensure-imgpkg.sh"
 "${TCE_REPO_PATH}/hack/ensure-deps/ensure-kbld.sh"
 "${TCE_REPO_PATH}/hack/ensure-deps/ensure-ytt.sh"

--- a/hack/ensure-deps/ensure-zip.sh
+++ b/hack/ensure-deps/ensure-zip.sh
@@ -1,0 +1,39 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script can be used to check your local development environment
+# for necessary dependencies used in TCE
+
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+TCE_CI_BUILD="${TCE_CI_BUILD:-""}"
+
+SUDO_CMD="sudo"
+if [[ "${TCE_CI_BUILD}" == "true" ]]; then
+  SUDO_CMD=""
+fi
+
+if [[ -z "$(command -v zip)" ]]; then
+  if [[ -f "/etc/redhat-release" ]]; then
+    ${SUDO_CMD} yum -y install zip
+  elif [[ "$(grep Ubuntu /etc/os-release)" != "" ]]; then
+    ${SUDO_CMD} apt-get -y install zip
+  else
+    echo "**** Please install zip before proceeding *****"
+    exit 1
+  fi
+fi
+if [[ -z "$(command -v unzip)" ]]; then
+  if [[ -f "/etc/redhat-release" ]]; then
+    ${SUDO_CMD} yum -y install unzip
+  elif [[ "$(grep Ubuntu /etc/os-release)" != "" ]]; then
+    ${SUDO_CMD} apt-get -y install unzip
+  else
+    echo "**** Please install unzip before proceeding *****"
+    exit 1
+  fi
+fi

